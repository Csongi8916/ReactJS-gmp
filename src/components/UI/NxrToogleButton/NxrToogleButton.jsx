import React, { useState } from 'react';

import './NxrToogleButton.scss';

export default function NxrToogleButton(props) {
  return (
    <div className="NxrToogle" >
      <span className="NxrToogle_ModeTitle">{props.title}</span>
      <div className="NxrToogle_Container">
        <button value={props.firstTitle}
          className={`NxrToogle_Btn ${
            (props.shearchParam === props.firstTitle)
              || (props.sortParam === props.firstTitle)
              ? 'NxrToogle_Btn-Active'
              : 'NxrToogle_Btn-Inactive'}`}
          onClick={(e) => props.click(e)}>
          {props.firstTitle}
        </button>
        <button value={props.secondTitle}
          className={`NxrToogle_Btn ${
            (props.shearchParam === props.secondTitle)
              || (props.sortParam === props.secondTitle)
              ? 'NxrToogle_Btn-Active'
              : 'NxrToogle_Btn-Inactive'}`}
          onClick={(e) => props.click(e)}>
          {props.secondTitle}
        </button>
      </div>
    </div>
  )
}
