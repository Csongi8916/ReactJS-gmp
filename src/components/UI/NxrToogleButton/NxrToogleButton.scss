$nrRedColor: #f65261;
$nrBlackColor: #232323;
$nrGrayColor: #424242;
$nrWhiteColor: #ffffff;

.NxrToogle {
  display: flex;
  align-items: center;

  .NxrToogle_ModeTitle {
    color: #fff;
    font-size: 16px;
    margin-right: 20px;
    text-transform: uppercase;
  }

  .NxrToogle_Container {
    display: flex;
    text-align: center;
    margin-left: 10px;
    cursor: pointer;
    color: #fff;
    font-family: Montserrat, Helvetica, sans-serif;
    font-size: 16px;
    text-transform: uppercase;

    .NxrToogle_Btn {
      border: 0;
      border-radius: 3%;
      padding: 15px 25px;
    }

    .NxrToogle_Btn-Active {
      background-color: $nrRedColor;
    }

    .NxrToogle_Btn-Inactive {
      background-color: rgba( $nrGrayColor, .7 );
    }

  }
}
