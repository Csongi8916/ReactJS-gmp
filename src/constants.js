export const SEARCH_BY = 'search by';
export const SORT_BY = 'sort by';

export const TITLE = 'title';
export const GENRE = 'genres';
export const RELEASED_DATE = 'released date';
export const RAITING = 'raiting';