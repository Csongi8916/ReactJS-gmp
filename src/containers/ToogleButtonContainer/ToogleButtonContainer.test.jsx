import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import NxrToogleButton from '../../components/UI/NxrToogleButton/NxrToogleButton';
import * as constants from '../../constants';

describe('ToogleButtonContainer Snapshpt tests', () => {

});

describe('ToogleButtonContainer Snapshpt tests:', () => {
  test('Should render correctly NxrToogleButton', () => {
    const tree = renderer.create(<NxrToogleButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});