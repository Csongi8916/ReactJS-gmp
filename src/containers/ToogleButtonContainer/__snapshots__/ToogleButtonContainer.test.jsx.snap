// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToogleButtonContainer Snapshpt tests: Should render correctly NxrToogleButton 1`] = `
<div
  className="NxrToogle"
>
  <span
    className="NxrToogle_ModeTitle"
  />
  <div
    className="NxrToogle_Container"
  >
    <button
      className="NxrToogle_Btn NxrToogle_Btn-Active"
      onClick={[Function]}
    />
    <button
      className="NxrToogle_Btn NxrToogle_Btn-Active"
      onClick={[Function]}
    />
  </div>
</div>
`;
